File: Laba1/Task7.cs
using Extensions;

namespace PZPK_Labas.Laba1
{
    public static class Task7
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 7. Написати програму, згідно умови завдання, використовуючи оператор циклу з параметром (оператор for).");

            int k = 9;
            int x;

            ConsoleInput.SafeIntInput("Введіть x", out x);

            Console.WriteLine($"Сума послідовності: {SumTask7(k, x)}");

            Console.ReadKey();
            Console.WriteLine();
        }

        private static double SumTask7(int k, int x)
        {
            double sum = 0;

            for (int n = 1; n < k; n++)
            {
                sum += Math.Pow(-1, n) * ((n + 1) / (Math.Pow(n * x, 2) + 1));
            }

            return sum;
        }
    }
}

File: Laba1-Python/Task7.py
import Extensions
import math

def Run():
    print("Завдання 7. Написати програму, згідно умови завдання, використовуючи оператор циклу з параметром (оператор for).")

    k = Extensions.SafeIntInput("Введіть k")
    x = Extensions.SafeIntInput("Введіть x")

    print(f"Сума послідовності: {SumTask7(k, x)}")

    input()
    print()


def SumTask7(k, x):
    sum = 0

    for n in range(1, int(k)):
        sum += pow(-1, n) * ((n + 1) / (pow(n * x, 2) + 1))

    return round(sum, 3)
